src/util/log.c

```diff
@@ -15,6 +16,56 @@
 #define unlikely(x)     (x)
 #endif
 
+/**
+ * Compares two strings byte by byte and ignores the
+ * character's case. Stops at the n-th byte of both
+ * strings.
+ * 
+ * This is basically a replacement of the POSIX-function
+ * _strncasecmp_. Since tpm2-tss is supposed to be compatible
+ * with ISO C99 and not with POSIX, _strncasecmp_ had to be
+ * replaced. This function creates lowercase representations
+ * of the strings to compare and calls the function _strncmp_
+ * on them.
+ *
+ * @param string1 The first of the two strings to compare
+ * @param string2 The second of the two strings to compare
+ * @param n The maximum number of bytes to compare
+ * @return 0 if both strings are equal (case insensitive),
+ *  an integer greater than zero if string1 is greater than
+ *  string 2 and an integer smaller than zero if string1 is
+ *  smaller than string2
+ *
+ */
+static int caseInsensitiveStrncmp(const char *string1, 
+        const char *string2, 
+        size_t n)
+{
+    if ((string1 == NULL) && (string2 == NULL)) {
+        return 0;
+    }
+    if ((string1 == NULL) && (string2 != NULL)) {
+        return -1;
+    }
+    if ((string1 != NULL) && (string2 == NULL)) {
+        return 1;
+    }
+
+    size_t string1_size = strlen (string1);
+    size_t string2_size = strlen (string2);
+    char lower_string1 [string1_size];
```

Comment:
If you would declare these as unsigned char there would be no need for casting at all.

Reply:
You mean the two arrays `lower_string1` and  `2`? They need to be char, otherwise `strncmp` won't work on them (signedness warnings that result in errors). The casting is also not due to signedness but due to `toLower` taking integer values as input and output and those lines cause size warnings if I don't cast here.

Reply:
Yes, but now there are 4 casts, 2 for each call to tolower(). With unsigned char only two would be needed. If we change strncmp() to memcmp(), then no cast is needed:

    unsigned char lower_string1 [string1_size];
    unsigned char lower_string2 [string2_size];

    for (size_t i = 0; i < string1_size; i++)
        lower_string1[i] = tolower(string1[i]);

    for (size_t i = 0; i < string2_size; i++)
        lower_string2[i] = tolower(string2[i]);

    return memcmp (lower_string1, lower_string2, n);

Reply:
Ah...now I get it. It's done and the commits are squashed.